src/coreclr/jit/gentree.cpp

```diff
@@ -24768,3 +24768,85 @@ bool GenTree::IsNeverNegative(Compiler* comp) const
     // TODO-Casts: extend IntegralRange to handle constants
     return IntegralRange::ForNode((GenTree*)this, comp).IsPositive();
 }
+
+//------------------------------------------------------------------------
+// IsNeverNegativeOne: returns true if the given tree is known to never be
+//                     be negative one. Only valid for integral types.
+//
+// Arguments:
+//    comp - Compiler object, needed for IsNeverNegative
+//
+// Return Value:
+//    true if the given tree is known to never be negative one
+//
+bool GenTree::IsNeverNegativeOne(Compiler* comp) const
```

Comment:
this helper is only needed for division it seems?

Reply:
It's only in one place, but it does help make things clean.